### PR TITLE
Add links to documentation for all commands

### LINKED ISSUE #1729: cli: add links to DVC's documentation in each command help string
```diff
❯ dvc metrics --help
usage: dvc metrics [-h] [-q | -v] {show,add,modify,remove} ...

A set of commands to add, manage, collect and display project metrics.
+
+ documentation: https://dvc.org/doc/commands-reference/metrics

positional arguments:
  {show,add,modify,remove}
                        Use dvc metrics CMD --help to display command-specific
                        help.
    show                Output metric values.
    add                 Tag file as a metric file.
    modify              Modify metric file options.
    remove              Remove files's metric tag.

optional arguments:
  -h, --help            show this help message and exit
  -q, --quiet           Be quiet.
  -v, --verbose         Be verbose.
```

### PR DESCRIPTION
**Pull request by J0**
Fix #1729 . It seems that there isn't any documentation for `dvc daemon` and `dvc pkg`  so I am omitting it for now.

### CODE REVIEW

**Comment by J0** [comment]
I have added documentation links to all the commands that I could find through `dvc --help`. Do let me know if there are any edits required. I'm not too sure where the help message for`dvc status` is though( I wasn't able to find it in `dvc/dvc/command/status.py`. Would be great if someone could point me in the right direction-- if not then I'll do a search tmrw

**Comment by shcheklein** [comment]
This is great, @J0 ! Thanks. I suggest I add a dvc shortener for these links - something like `help.dvc.org/commit` -> `dvc.org/doc/command-reference/commit`. The main goal is to have a level of indirection we can control on the dvc.org website level. It's hard to always keep doc/* links up to date and I think it's not the best idea to depend on this. Give a bit of time to setup this redirect. What do you think? :)

**Comment by shcheklein** [comment]
@J0 man.dvc.org is alive. Try this: https://man.dvc.org/metrics or https://man.dvc.org/metrics-show

**Comment by shcheklein** [comment]
@J0 take a look at `/dvc/command/data_sync.py` for status command.

**Comment by shcheklein** [comment]
@J0 also, let's add help messages for subcommands as well - for example - https://man.dvc.org/metrics-show for `dvc metrics show`.

File: dvc/command/move.py [hunk 1]
```diff
@@ -19,12 +19,14 @@ def run(self):
 
 
 def add_parser(subparsers, parent_parser):
+    MOVE_HELP = "Rename or move a DVC controlled data file or a directory.\n\
+documentation: documentation: https://dvc.org/doc/commands-reference/move"
```
Comment (shcheklein) [inline dvc/command/move.py]: duplicate `documentation:`

**Comment by dmpetrov** [comment]
It might be better to squash 32 commits into 1-10. Otherwise it might break our beautiful contributors-graph by creating a spike and changing the graph scale :)

```
$ git rebase -i HEAD~30
# follow instructions
$ git push -f
```

See: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by shcheklein** [comment]
@dmpetrov there is a button for that. @efiop will do Squash and merge when PR is ready. No need to redo the history in the PR itself.

**Comment by J0** [comment]
Hey @shcheklein sounds good to me -- I'll get to it in about 10 minutes :)

File: dvc/command/move.py (hunk 1 above)
Comment (J0) [reply dvc/command/move.py]: removed duplicate `documentation`

**Comment by shcheklein** [comment]
@J0 no rush, thanks! I think we need to keep https:// prefix. There are some terminals that can make them clickable automatically. I'm not sure about man.dvc.org without the prefix. I would double check that.

**Comment by shcheklein** [comment]
saying that, it looks like Mac OS terminal can recognize the shorter version - man.dvc.org/metrics ... what about other environments - @efiop @mroutis @pared ?

File: dvc/command/add.py [hunk 2]
```diff
@@ -26,7 +26,8 @@ def run(self):
 
 
 def add_parser(subparsers, parent_parser):
-    ADD_HELP = "Add files/directories to dvc."
+    ADD_HELP = "Add files/directories to dvc. \n\
```
Comment (efiop) [inline dvc/command/add.py]: Let's reformat all of these this way:
```
    ADD_HELP = (
        "Add files/directories to dvc. \n"
        "documentation: https://man.dvc.org/add"
    )
```

so it is unified everywhere and we don't mix \ into all of these.

Comment (J0) [reply dvc/command/add.py]: Ok done

**Pull request merged**